src/argilla/server/apis/v0/handlers/users.py

```diff
@@ -95,9 +95,12 @@ async def create_user(
     if user is not None:
         raise EntityAlreadyExistsError(name=user_create.username, type=User)
 
-    user = await accounts.create_user(db, user_create)
-    await user.awaitable_attrs.workspaces
+    try:
+        user = await accounts.create_user(db, user_create)
+    except Exception as e:
+        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(e))
 
+    await user.awaitable_attrs.workspaces
```

Comment:
Just to confirm, if no `workspaces` are provided this won't add any overhead right? Couldn't we also add `if user_create.workspaces is not None` right before?